### PR TITLE
fix: [PL-37900]: Add non connected delegates to selection log

### PR DESCRIPTION
**Pull request by jennyjames21**
## Describe your changes
Summary:
Add non active delegate info to selection log.

Test
-PR Env testing with few delegates disconnected

## Checklist
- [x] I've documented the changes in the PR description.
- [x] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

## [Latest PR Check Triggers](https://github.com/harness/harness-core/blob/develop/.github/pull_request_template.md)

<details>
  <summary>PR Check triggers</summary>
You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ut0, ut1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger uts`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
- Module Dependency Check: `trigger mdc`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/48203)

### CODE REVIEW

**Review by mdmilic — APPROVED** [review]
> Reviewed 3 of 3 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/48203)* status: Waiting on team(s): del-team.

File: 400-rest/src/main/java/software/wings/service/impl/AssignDelegateServiceImpl.java [hunk 1]
```diff
@@ -1106,9 +1107,20 @@ public boolean canAssignTaskV2(String delegateId, DelegateTask task) {
...
   @Override
-  public List<Delegate> fetchActiveDelegates(String accountId) {
-    List<Delegate> accountDelegates = getAccountDelegates(accountId);
+  public List<Delegate> fetchActiveDelegates(DelegateTask delegateTask) {
+    List<Delegate> accountDelegates = getAccountDelegates(delegateTask.getAccountId());
     long oldestAcceptableHeartBeat = currentTimeMillis() - MAX_DELEGATE_LONG_LAST_HEARTBEAT;
+    long delegateConnectedCheck = currentTimeMillis() - MAX_DELEGATE_LAST_HEARTBEAT;
+    List<String> nonConnectedDelegates = accountDelegates.stream()
+                                             .filter(delegate
+                                                 -> delegate.getStatus() == DelegateInstanceStatus.ENABLED
```
Comment (raghuAtWings) [inline 400-rest/src/main/java/software/wings/service/impl/AssignDelegateServiceImpl.java:1116]: what does enabled mean?

**Comment by jennyjames21** [comment]
*[`400-rest/src/main/java/software/wings/service/impl/AssignDelegateServiceImpl.java` line 1116 at r2](https://reviewable.io/reviews/harness/harness-core/48203#gh-1204751866:-NWEaqkr1b9kUrafQOKE:b-peu01b) ([raw file](https://github.com/harness/harness-core/blob/dc5204f03edf6e85e9b7e202c2190b7d90699724/400-rest/src/main/java/software/wings/service/impl/AssignDelegateServiceImpl.java#L1116)):*
<details><summary><i>Previously, raghuAtWings (Raghvendra Singh) wrote…</i></summary><blockquote>

what does enabled mean? 
</blockquote></details>

Other status is DELETED, WAITING FOR APPROVAL.

**Comment by jennyjames21** [comment]
*[`400-rest/src/main/java/software/wings/service/impl/AssignDelegateServiceImpl.java` line 1116 at r2](https://reviewable.io/reviews/harness/harness-core/48203#gh-1204751866:-NWEbKQ6CaIM1Twhll43:b-b4a7u3) ([raw file](https://github.com/harness/harness-core/blob/dc5204f03edf6e85e9b7e202c2190b7d90699724/400-rest/src/main/java/software/wings/service/impl/AssignDelegateServiceImpl.java#L1116)):*
<details><summary><i>Previously, jennyjames21 (Jenny James) wrote…</i></summary><blockquote>

Other status is DELETED, WAITING FOR APPROVAL.
</blockquote></details>

**Comment by jennyjames21** [comment]
*[`400-rest/src/main/java/software/wings/service/impl/AssignDelegateServiceImpl.java` line 1116 at r2](https://reviewable.io/reviews/harness/harness-core/48203#gh-1204751866:-NWEbM9r993ExoKw9Rqv:b-lxjsro) ([raw file](https://github.com/harness/harness-core/blob/dc5204f03edf6e85e9b7e202c2190b7d90699724/400-rest/src/main/java/software/wings/service/impl/AssignDelegateServiceImpl.java#L1116)):*
<details><summary><i>Previously, jennyjames21 (Jenny James) wrote…</i></summary><blockquote>


</blockquote></details>

prod^^

File: 400-rest/src/main/java/software/wings/service/impl/AssignDelegateServiceImpl.java (hunk 1 above)
Comment (jennyjames21) [reply 400-rest/src/main/java/software/wings/service/impl/AssignDelegateServiceImpl.java]: removed that check

**Comment by jennyjames21** [comment]
trigger uts

**Comment by jennyjames21** [comment]
trigger uts

**Review by raghuAtWings — APPROVED** [review]

**Comment by jennyjames21** [comment]
trigger messagecheck

**Pull request merged**